### PR TITLE
Aliyun node provider

### PR DESCRIPTION
**Pull request by zhuangzhuang131419**
## Why are these changes needed?
Add a node provider for aliyun cluster.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/cloud.rst [hunk 1]
```diff
@@ -113,6 +113,28 @@ AWS/GCP/Azure
             # Tear down the cluster.
             $ ray down ray/python/ray/autoscaler/gcp/example-full.yaml
 
+    .. group-tab:: Aliyun
+
+        First, you need to configure your aliyun account. You need to set access key and access key secret in yaml config file.
+
+        Once above done, you should be ready to launch your cluster. The provided `ray/python/ray/autoscaler/aliyun/example-full.yaml <https://github.com/alipay/ray/blob/zhuang_aliyun/python/ray/autoscaler/aliyun/example-full.yaml>`__ cluster config file will create a small cluster with an ecs.n4.large head node (on-demand) configured to autoscale up to two ecs.n4.large.
```
Comment (chenk008) [inline doc/source/cluster/cloud.rst:120]: url用相对路径

Comment (chenk008) [reply doc/source/cluster/cloud.rst:120]: autoscale up to two  `ecs.n4.large.`  这个要对着example-full来写

File: python/ray/autoscaler/_private/aliyun/node_provider.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+            region=provider_config["region"],
+            max_retries=BOTO_MAX_RETRIES,
+        )
+
+        self.acs_fail_fast = AcsClient(
+            access_key=provider_config["access_key"],
+            access_key_secret=provider_config["access_key_secret"],
+            region=provider_config["region"],
+            max_retries=1,
+        )
```
Comment (chenk008) [inline python/ray/autoscaler/_private/aliyun/node_provider.py:47]: 这个用不到先去掉吧

File: python/ray/autoscaler/aliyun/defaults.yaml [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    key_name: ray
+
+
+# How Ray will ç with newly launched nodes.
+auth:
+    ssh_user: root
+# By default Ray creates a new private keypair, but you can also use your own.
+# If you do so, make sure to also set "KeyName" in the head and worker node
+# configurations below.
+    ssh_private_key: ~/.ssh/id_rsa
```
Comment (chenk008) [inline python/ray/autoscaler/aliyun/defaults.yaml:40]: 这里补充一下public key

File: python/ray/autoscaler/aliyun/defaults.yaml [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+# List of shell commands to run to set up nodes.
+setup_commands:
+    - sudo apt-get update
+    # Install Anaconda.
+    - wget https://repo.continuum.io/archive/Anaconda3-5.0.1-Linux-x86_64.sh || true
+    - bash Anaconda3-5.0.1-Linux-x86_64.sh -b -p $HOME/anaconda3 || true
+    - echo 'export PATH="$HOME/anaconda3/bin:$PATH"' >> ~/.bashrc
+    # Install Ray
+    - pip install -U ~/dist/ray-2.0.0.dev0-cp36-cp36m-linux_x86_64.whl
```
Comment (chenk008) [inline python/ray/autoscaler/aliyun/defaults.yaml:135]: pip install -U ray

File: python/ray/autoscaler/aliyun/defaults.yaml [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    cidr_block: 172.16.0.0/24
+    # Whether to allow node reuse. If set to False, nodes will be terminated
+    # instead of stopped.
+    cache_stopped_nodes: True # If not present, the default is True.
+    access_key:
+    access_key_secret:
+    key_name: default
+
+
+# How Ray will ç with newly launched nodes.
```
Comment (raulchen) [inline python/ray/autoscaler/aliyun/defaults.yaml:36]: typo `ç`?

**Pull request closed without merging**